### DISCUSSION TITLE
Firebase Dynamic Link is not working in FB Messenger and Line Chat App

### DISCUSSION THREAD

**Discussion by imNKsid** [invertase/react-native-firebase#6289] [Q&A]
Hi all 

I'm meeting with a weird bug, can anyone help me with this -
In IOS, when you share a class via FB messenger and when a user clicks on the link -
First time, it will take you to App Store saying the app is no longer available.
Second time, it will take you to the Home screen of the App (not to the class).
I've checked the logs, the link which we're sending is fine but the link which is taking the user back to the app is coming as NULL.
And this particular thing is happening only in IOS FB messenger.

The packages which I've used are -
```json
    "@react-native-firebase/analytics": "^14.11.0",
    "@react-native-firebase/app": "^14.11.0",
    "@react-native-firebase/auth": "^14.11.0",
    "@react-native-firebase/crashlytics": "^14.11.0",
    "@react-native-firebase/dynamic-links": "^14.11.0",
    "@react-native-firebase/messaging": "^14.11.0",
    "@react-native-firebase/storage": "^14.11.0",
```

The Firebase Dynamic Link which I'm sending to FB is like this - "https://<APP_NAME>.page.link/qYJtE7RDqitreexo7" . But the link which is coming back to the app in "handleDynamicLink" is like this - "https://<APP_NAME>.page.link/share". It should be like this - "https://<APP_NAME>.page.link/share?qYJtE7RDqitreexo7".

These are the functions which I'm using in the ComponentDidMount() -

```javascript
    dynamicLinks().getInitialLink().then(this.handleDynamicLink);
    dynamicLinks().onLink(this.handleDynamicLink);   

 handleDynamicLink = async link => {
    console.log('PENDING Guide Home', link);
    if (link?.url) {
      const arr = link.url.split('/');
      const temp = arr[arr.length - 1];
      const arr2 = temp.split('?');
      const action = arr2[0];
      let q = arr2[1].split('=')[1];
      q = decodeURIComponent(q);
      let decodedBase64 = base64.decode(q);
      let data = JSON.parse(decodedBase64);

      console.log('LINK DATA ', this.props);
      if (action === 'share' && data.classId && this.props.isSignIN) {
        console.log('NAVOGATING');
        // await this.props.getSingleClass(data.classId);
        this.props.navigation.navigate('ClassDetails', {
          selectedClass: data.classId,
        });
      }
    }
  };
```

**Comment by mikehardy** [comment]
I reformatted your question with code formatting markdown chunks for the json and javascript, much easier to read (triple-backticks and code language modifier is the trick)

You may be encoding it incorrectly during construction, hard to say
https://firebase.google.com/docs/dynamic-links/debug


Also, I implemented this just recently but you have it with 14.11.0:

https://firebase.google.com/docs/dynamic-links/debug#ios_self-diagnostic_tool
https://rnfirebase.io/reference/dynamic-links#performDiagnostics

...may help?

Be careful with the encoding, you don't show how you do it so your snippet is not complete, you need to make sure it is urlencoded

**Reply by imNKsid** [reply]
It's not working only for FB and FB Messenger.

**Reply by mikehardy** [reply]
You will probably need to take that up with firebase-ios-sdk then in their repo, if it's ios-specific, and works in general, then it's "working" here. You're building and using links correctly. If there's a mal-interaction between firebase-ios-sdk and FB products specifically then google and/or meta will have to work that out, but we won't have "jurisdiction" (any ability to control it) here
